Code/GraphMol/SynthonSpaceSearch/Synthon.h

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright (C) David Cosgrove 2024.
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#ifndef RDKIT_REAGENT_H
+#define RDKIT_REAGENT_H
+
+#include <string>
+
+#include <DataStructs/ExplicitBitVect.h>
+#include <GraphMol/ROMol.h>
+
+namespace RDKit {
+class Atom;
+
+namespace SynthonSpaceSearch {
+
+// This class holds a Synthon that will be part of a SynthonSet.
+class RDKIT_SYNTHONSPACESEARCH_EXPORT Synthon {
+ public:
+  Synthon() = default;
+  Synthon(const std::string &smi, const std::string &id);
+  Synthon(const Synthon &other);
+  Synthon(Synthon &&other) noexcept;
```

Comment:
Your implementation of this (and the move-assignment operator) are identical to what the default would be, so you might as well use that:
```suggestion
  Synthon(Synthon &&other) = default;
```